src/tree_cmd.cpp

```diff
@@ -803,10 +803,16 @@ static void TileLoopTreesAlps(TileIndex tile)
 	MarkTileDirtyByTile(tile);
 }
 
+/*
+ * Check if trees on this tile can spread, according to biome and game settings.
+ * 
+ * In most climates, the setting that decides is to "grow and spread everywhere."
+ * If it is a tropical game, then it must be in the rainforest and have a setting allowing it to spread there.
+ */
 static bool CanPlantExtraTrees(TileIndex tile)
 {
-	return ((_settings_game.game_creation.landscape == LandscapeType::Tropic && GetTropicZone(tile) == TROPICZONE_RAINFOREST) ?
-		(_settings_game.construction.extra_tree_placement == ETP_SPREAD_ALL || _settings_game.construction.extra_tree_placement == ETP_SPREAD_RAINFOREST) :
+	return ((_settings_game.game_creation.landscape == LandscapeType::Tropic) ?
+		(GetTropicZone(tile) == TROPICZONE_RAINFOREST && (_settings_game.construction.extra_tree_placement == ETP_SPREAD_ALL || _settings_game.construction.extra_tree_placement == ETP_SPREAD_RAINFOREST)) :
```

Comment:
Your code isn't quite right; `TROPICZONE_NORMAL` is treated like desert.

The ternary is hard to read here, I'd suggest expanding it and explicitly handling desert tiles. Here's how I'd do it (I've tested this but would encourage you to also test it yourself):

```suggestion
/* Desert and rainforest trees need special handling. */
if (_settings_game.game_creation.landscape == LandscapeType::Tropic) {
	switch (GetTropicZone(tile)) {
		case TROPICZONE_DESERT:
			/* Cacti never spread. */
			return false;
		case TROPICZONE_RAINFOREST:
			return (_settings_game.construction.extra_tree_placement == ETP_SPREAD_ALL || _settings_game.construction.extra_tree_placement == ETP_SPREAD_RAINFOREST);
		default:
			return _settings_game.construction.extra_tree_placement == ETP_SPREAD_ALL;
	}
}

return (_settings_game.construction.extra_tree_placement == ETP_SPREAD_ALL);
```